pennylane/data/dataset.py

```diff
@@ -73,25 +75,57 @@ class Dataset(ABC):
         + (1) [Z1]
     """
 
-    def __init__(self, dtype=None, **kwargs):
-        self.dtype = dtype
+    _standard_argnames = ["data_type", "folder", "attr_prefix"]
+
+    def __std_init__(self, data_type, folder, attr_prefix):
+        """Constructor for standardized datasets."""
+        self._dtype = data_type
+        self._folder = folder.rstrip("/")
+        self._prefix = os.path.join(self._folder, attr_prefix.rstrip("/")) + "_{}.dat"
         self.__doc__ = ""
+
+        self._fullfile = self._prefix.format("full")
+        if not os.path.exists(self._fullfile):
+            self._fullfile = None
+
+        for f in glob(self._folder + "/*.dat"):
+            data = self._read_file(f)
+            for attr, value in data.items():
+                setattr(self, f"{attr}", value)
+
+    def __base_init__(self, **kwargs):
+        """Constructor for user-defined datasets."""
         for key, val in kwargs.items():
             setattr(self, key, val)
 
-    def __eq__(self, __o):
-        return self.__dict__ == __o.__dict__
+    def __init__(self, *args, standard=False, **kwargs):
```

Comment:
I'd rather not, because for users, there are no explicit arguments! Those are only for the "standard" Datasets, which should only be created internally by us.

Reply:
chatted offline, updated the docstring. will need validation on the website before merging